cylc/rose/entry_points.py

```diff
@@ -13,320 +13,74 @@
-    if not rose_config_exists(srcdir, opts):
-        return False
-    srcdir, rundir = paths_to_pathlib([srcdir, rundir])
-    results = {}
-    copy_config_file(srcdir=srcdir, rundir=rundir)
-    results['record_install'] = record_cylc_install_options(
-        srcdir=srcdir, opts=opts, rundir=rundir
-    )
-    results['fileinstall'] = rose_fileinstall(
-        srcdir=srcdir, opts=opts, rundir=rundir
-    )
-    # Finally dump a log of the rose-conf in its final state.
-    if results['fileinstall']:
-        dump_rose_log(rundir=rundir, node=results['fileinstall'])
-
-    return results
-
-
-def get_rose_vars(srcdir=None, opts=None):
-    """Load template variables from Rose suite configuration.
-
-    Loads the Rose suite configuration tree from the filesystem
-    using the shell environment.
-
-    Args:
-        srcdir(pathlib.Path):
-            Path to the Rose suite configuration
-            (the directory containing the ``rose-suite.conf`` file).
-        opts:
-            Options object containing specification of optional
-            configuarations set by the CLI.
-
-    Returns:
-        dict - A dictionary of sections of rose-suite.conf.
-        For each section either a dictionary or None is returned.
-        E.g.
-            {
-                'env': {'MYVAR': 42},
-                'empy:suite.rc': None,
-                'jinja2:suite.rc': {
-                    'myJinja2Var': {'yes': 'it is a dictionary!'}
-                }
-            }
-    """
-    # Set up blank page for returns.
-    config = {
-        'env': {},
-        'template_variables': {},
-        'templating_detected': None
-    }
-
-    # Return a blank config dict if srcdir does not exist
-    if not rose_config_exists(srcdir, opts):
-        if (
-            getattr(opts, "opt_conf_keys", None)
-            or getattr(opts, "defines", None)
-            or getattr(opts, "rose_template_vars", None)
-        ):
-            raise NotARoseSuiteException()
-        return config
-
-    # Check for definitely invalid defines
-    if opts and hasattr(opts, 'defines'):
-        invalid_defines_check(opts.defines)
-
-    # Load the raw config tree
-    config_tree = rose_config_tree_loader(srcdir, opts)
-    deprecation_warnings(config_tree)
-
-    # Extract templatevars from the configuration
-    get_rose_vars_from_config_node(
-        config,
-        config_tree.node,
-        os.environ
-    )
-
-    # Export environment vars
-    for key, val in config['env'].items():
-        os.environ[key] = val
 
-    return config
+if TYPE_CHECKING:
+    from cylc.flow.option_parsers import Values
 
 
-def record_cylc_install_options(
-    rundir=None,
-    opts=None,
-    srcdir=None,
-):
-    """Create/modify files recording Cylc install config options.
+def pre_configure(srcdir: Path, opts: 'Values') -> dict:
+    """Run before the Cylc configuration is read."""
+    if not srcdir:
+        # not sure how this could happen
```

Comment:
I'm not either - why did you add it?

Reply:
This case was handled by the code before, I didn't get around to removing it.

Reply:
If `srcdir` is really a `Path`, then it can never be falsy